builtin/credential/aws/client.go

```diff
@@ -130,7 +132,12 @@ func (b *backend) getClientConfig(ctx context.Context, s logical.Storage, region
 		if err != nil {
 			return nil, err
 		}
-		assumedCredentials := stscreds.NewCredentials(sess, stsRole, func(p *stscreds.AssumeRoleProvider) { p.ExternalID = aws.String(externalID) })
+		var assumedCredentials *credentials.Credentials
+		if externalID != "" {
+			assumedCredentials = stscreds.NewCredentials(sess, externalID, func(p *stscreds.AssumeRoleProvider) { p.ExternalID = aws.String(externalID) })
+		} else {
+			assumedCredentials = stscreds.NewCredentials(sess, externalID)
```

Comment:
I see that in the old code, `stsRole` is being passed as the second parameter 5o `NewCredentials` but here it is changed to `externalID` for both if statement cases. Should it still be `stsRole`?

Reply:
yeah I did a bad thing and let the autocomplete handle it when i reformatted the if-statement.